portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +26,16 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = request.getParameter("name");
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+
+    if (name == null) {
+        response.getWriter().println("<h1>Oops, I don't know what your name is.</h1>");
+    } else {
+        response.getWriter().println("<h1>Hello " + name + "!</h1>");
```

Comment:
Even though there aren't any obvious dangers of exposing this site to a reflective XSS attack, I still think it's probably better to avoid it here if possible. Could you add a quick check before using name to ensure it only contains alphanumeric characters? It looks like this can be accomplished using regex, as seen here: https://www.thecrazyprogrammer.com/2015/07/how-to-check-string-is-alphanumeric-in-java.html

Reply:
Yeah it's probably good practice to make it safe anyways. I went and added some sanitization using regex as you suggested.